amaranth/hdl/__init__.py

```diff
@@ -6,8 +6,8 @@
 from ._ast import IOValue, IOPort
 from ._dsl import Module
 from ._cd import DomainError, ClockDomain
-from ._ir import AlreadyElaborated, UnusedElaboratable, Elaboratable, DriverConflict, Fragment
-from ._ir import Instance, IOBufferInstance
+from ._ir import AlreadyElaborated, UnusedElaboratable, Elaboratable, DriverConflict, DomainRequirementFailed
```

Comment:
Should this be exported? I don't want people catching this exception without importing it from a private module.

Reply:
Right, good point.